examples/torch/dqn_atari.py

```diff
@@ -0,0 +1,195 @@
+from garage.sampler import FragmentWorker, LocalSampler
+from garage.torch import set_gpu_mode
+from garage.torch.algos import DQN
+from garage.torch.policies import DiscreteQFArgmaxPolicy
+from garage.torch.q_functions import DiscreteCNNQFunction
+from garage.trainer import Trainer
+
+hyperparams = dict(n_epochs=500,
+                   steps_per_epoch=20,
+                   sampler_batch_size=500,
+                   lr=1e-4,
+                   discount=0.99,
+                   min_buffer_size=int(1e4),
+                   n_train_steps=125,
+                   target_update_freq=2,
+                   buffer_batch_size=32,
+                   max_epsilon=1.0,
+                   min_epsilon=0.01,
+                   decay_ratio=0.1,
+                   buffer_size=int(1e4),
+                   hidden_sizes=(512, ),
+                   hidden_channels=(32, 64, 64),
+                   kernel_sizes=(8, 4, 3),
+                   strides=(4, 2, 1),
+                   clip_gradient=10)
+
+
+@click.command()
+@click.argument('env', type=str)
+@click.option('--seed', default=24)
+@click.option('--n', type=int, default=psutil.cpu_count(logical=False))
+@click.option('--buffer_size', type=int, default=None)
+@click.option('--max_episode_length', type=int, default=None)
+def main(env=None,
+         seed=24,
+         n=psutil.cpu_count(logical=False),
+         buffer_size=None,
+         max_episode_length=None):
+    """Wrapper to setup the logging directory.
+
+    Args:
+        env (str): Name of the atari environment, can either be the prefix
+            or the full name. For example, this can either be 'Pong' or
+            'PongNoFrameskip-v4'. If the former is used, the env used will be
+            `env` + 'NoFrameskip-v4'.
+        seed (int): Seed to use for the RNG.
+        n (int): Number of workers to use. Defaults to the number of CPU cores
+            available.
+        buffer_size (int): size of the replay buffer in transitions. If None,
+            defaults to hyperparams['buffer_size']. This is used by the
+            integration tests.
+        max_episode_length (int): Max length of an episode. If None, defaults
+            to the timelimit specific to the environment. Used by integration
+            tests.
+    """
+    if '-v' not in env:
+        env += 'NoFrameskip-v4'
+    logdir = 'data/local/experiment/' + env
+
+    if buffer_size is not None:
+        hyperparams['buffer_size'] = buffer_size
+
+    dqn_atari(dict(log_dir=logdir),
+              env=env,
+              seed=seed,
+              n_workers=n,
+              max_episode_length=max_episode_length,
+              **hyperparams)
+
+
+# pylint: disable=unused-argument
+@wrap_experiment(snapshot_mode='gap_overwrite', snapshot_gap=30)
+def dqn_atari(ctxt=None,
+              env=None,
+              seed=24,
+              n_workers=psutil.cpu_count(logical=False),
+              max_episode_length=None,
+              **kwargs):
```

Comment:
please do not use **kwargs in experiment launchers. it circumvents the ideas being wrap_experiment

Reply:
So the idea behind this and the `hyperparams` dict is that it makes life easier when comparing experiments. Passing the hyperparms as a dict to the launcher function logs them into`variant.json` which is helpful later when debugging runs and comparing hyperparams.